include/avif/avif.h

```diff
@@ -52,7 +52,7 @@ extern "C" {
     (defined(__STDC_VERSION__) && __STDC_VERSION__ >= 202311L)
 #define AVIF_NODISCARD [[nodiscard]]
 #else
-#if defined(__clang__) && defined(__has_attribute)
+#if defined(__clang__) && defined(__has_attribute) && ((__clang_major__ << 8) | __clang_minor__) >= ((3 << 8) | 9)
```

Comment:
Vincent: Thank you for using finding the first version of clang that supports the `warn_unused_result` attribute.

It is strange that `__has_attribute(warn_unused_result)` is not working correctly for clang 3.4.2.

Optional: `((3 << 8) | 9)` can be written as `0x309` or `0x0309`.

Reply:
> Vincent: Thank you for using finding the first version of clang that supports the `warn_unused_result` attribute.
> 
> It is strange that `__has_attribute(warn_unused_result)` is not working correctly for clang 3.4.2.
> 

I think `__has_attribute` is working, but earlier versions of clang didn't allow the attribute to be used on enums (gcc still doesn't I believe). I didn't think adding it to enums was strictly necessary.

> Optional: `((3 << 8) | 9)` can be written as `0x309` or `0x0309`.

Reply:
James: You are right. I forgot what the bug was about. Thanks.

Yes, another solution would be to individually mark the public functions that return `avifResult` as `AVIF_NODISCARD`.

Vincent: I suggest adding a comment to explain the clang version >= 3.9 test is to check whether the `warn_unused_result` attribute can be used on enums. This is not obvious.

Reply:
Comment added in https://github.com/AOMediaCodec/libavif/pull/1954